src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -177,9 +177,26 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     bindingContext.ModelState.AddModelError(modelBindingKey, message);
                 }
             }
-            catch (Exception ex)
+            catch (InputFormatterException InputFormatterException)
             {
-                bindingContext.ModelState.AddModelError(modelBindingKey, ex, bindingContext.ModelMetadata);
+                bindingContext.ModelState.AddModelError(modelBindingKey, InputFormatterException, bindingContext.ModelMetadata);
```

Comment:
@Eilon - we should also make sure that we document and **DO**  allow these errors to return to the clients. 

We have a big problem today about how to get an appropriate and **safe** error message when a formatter fails to read the data. We should put a stake in the ground that `InputFormatterException` is always safe.

@kichalla the `AddModelError` code has a whitelist of exception types, can we add to that list?